org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenBuildSupport.java

```diff
@@ -26,23 +41,87 @@
  */
 public class MavenBuildSupport implements IBuildSupport {
 
+	private static final String POM_SERLIZATION_FILE_NAME = ".pom-digests";
+
+	private static final File SERIALIZATION_FILE;
+	private static Map<String, String> pomDigests;
+
+	static {
+		File workspaceFile = ResourcesPlugin.getWorkspace().getRoot().getRawLocation().makeAbsolute().toFile();
+		SERIALIZATION_FILE = new File(workspaceFile, POM_SERLIZATION_FILE_NAME);
+		if (SERIALIZATION_FILE.exists()) {
+			pomDigests = deserializePomDigests();
+		} else {
+			pomDigests = new HashMap<>();
+		}
+	}
+
+	private IProjectConfigurationManager configurationManager;
+
+	public MavenBuildSupport() {
+		this.configurationManager = MavenPlugin.getProjectConfigurationManager();
+	}
+
+	public MavenBuildSupport(IProjectConfigurationManager configurationManager) {
+		this.configurationManager = configurationManager;
+	}
+
 	@Override
 	public boolean applies(IProject project) {
 		return ProjectUtils.isMavenProject(project);
 	}
 
 	@Override
-	public void update(IProject project, IProgressMonitor monitor) throws CoreException {
```

Comment:
we might as well keep the  `update(IProject project, IProgressMonitor monitor)` flavour, to avoid the need to change to the more verbose `update(project, false, monitor)` version all over the place. Needs javadoc explaining the 2 signatures